core/controllers/acl_decorators.py

```
@@ -2753,6 +2753,84 @@ def test_can_accept_suggestion(
+        to accept translations andthe user who created a particular
+        translation are allowed to perform this action.
+
+    Raises:
+        NotLoggedInException. The user is not logged in.
+        UnauthorizedUserException. The user does not have credentials to
+            edit this suggestion.
+        InvalidInputException. The submitted suggestion id is not valid.
+        PageNotFoundException. A suggestion is not found with the given
+            suggestion id.
+    """
+    def test_can_update_suggestion(
+            self, suggestion_id, **kwargs):
+        """Returns a handler to test whether a
+        suggestion can be updated based on the user roles.
+
+        Args:
+            suggestion_id: str. The suggestion id.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            function. The handler for updating a
+            suggestion.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user does not have credentials to
+                edit this suggestion.
+            InvalidInputException. The submitted suggestion id is not valid.
+            PageNotFoundException. A suggestion is not found with the given
+                suggestion id.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+        user_actions = user_services.get_user_actions_info(
+            self.user_id
+        ).actions
+        if role_services.ACTION_ACCEPT_ANY_SUGGESTION in user_actions:
+            return handler(self, suggestion_id, **kwargs)
+
+        if len(suggestion_id.split('.')) != 3:
+            raise self.InvalidInputException(
+                'Invalid format for suggestion_id.'
+                ' It must contain 3 parts separated by \'.\'')
+
+        suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
+
+        if suggestion is None:
+            raise self.PageNotFoundException
```

Should we raise an exception message i.e, Invalid suggestion id? (Not sure whether we do it other places)

---

Most of the `PageNotFoundException` within the `acl_decorators.py` has been added in the same way. They have been raised w/o an exception message.